### PR TITLE
App was created

### PR DESCRIPTION
**Pull request by Artem-user-19**

### CODE REVIEW

File: todo/forms.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+from django import forms
+from .models import Task, Tag
```
Comment (viktoria-rybenchuk) [inline todo/forms.py]: [fix imports ](https://peps.python.org/pep-0008/#imports)

File: todo/urls.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    path('task/edit/<int:pk>/', TaskUpdateView.as_view(), name='edit_task'),
+    path('task/toggle/<int:pk>/', toggle_task_status, name='toggle_task'),
+
+    # Tag URLs
+    path('tags/', TagListView.as_view(), name='tag_list'),
+    path('tag/add/', TagCreateView.as_view(), name='add_tag'),
+    path('tag/edit/<int:pk>/', TagUpdateView.as_view(), name='edit_tag'),
+    path('tag/delete/<int:pk>/', TagDeleteView.as_view(), name='delete_tag'),
+    path('task/delete/<int:pk>/', TaskDeleteView.as_view(), name='delete_task'),
+]
```
Comment (viktoria-rybenchuk) [inline todo/urls.py]: add a blank line

File: todo/urls.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
+from django.urls import path
+from .views import (
+    TaskListView, TaskCreateView, TaskUpdateView, toggle_task_status,
+    TagListView, TagCreateView, TagUpdateView, TagDeleteView, TaskDeleteView
+)
+
+urlpatterns = [
+    # Task URLs
```
Comment (viktoria-rybenchuk) [inline todo/urls.py]: remove redundant comment

File: todo/urls.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    path('', TaskListView.as_view(), name='home'),
+    path('task/add/', TaskCreateView.as_view(), name='add_task'),
+    path('task/edit/<int:pk>/', TaskUpdateView.as_view(), name='edit_task'),
+    path('task/toggle/<int:pk>/', toggle_task_status, name='toggle_task'),
+
+    # Tag URLs
```
Comment (viktoria-rybenchuk) [inline todo/urls.py]: the same here

File: todo/urls.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
+from django.urls import path
+from .views import (
+    TaskListView, TaskCreateView, TaskUpdateView, toggle_task_status,
+    TagListView, TagCreateView, TagUpdateView, TagDeleteView, TaskDeleteView
+)
```
Comment (viktoria-rybenchuk) [inline todo/urls.py]: place each import on a new line and use absolute imports

File: todo/views.py [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    template_name = 'templates/task_form.html'
+    success_url = reverse_lazy('home')
+
+
+def toggle_task_status(request, pk):
+    task = get_object_or_404(Task, pk=pk)
+    task.is_done = not task.is_done
+    task.save()
+    return redirect('home')
+
```
Comment (viktoria-rybenchuk) [inline todo/views.py]: use class base view

**Review by viktoria-rybenchuk — CHANGES_REQUESTED** [review]
Remove db.sqlite3 and. pychache from the repository

File: todo/forms.py [hunk 7]
```diff
@@ -0,0 +1,26 @@
+from django import forms
+
+from .models import Task, Tag
```
Comment (pavlopro) [inline todo/forms.py]: use absolute import instead of relative

File: todo/urls.py [hunk 8]
```diff
@@ -0,0 +1,25 @@
+from django.urls import path
```
Comment (pavlopro) [inline todo/urls.py:1]: Add blank line between different import groups

use absolute import instead of relativ

File: todo/views.py [hunk 9]
```diff
@@ -0,0 +1,71 @@
+from django.urls import reverse_lazy
+from django.views.generic import ListView, CreateView, UpdateView, DeleteView
+from django.views.generic.base import View
+from django.shortcuts import redirect
+from django.shortcuts import get_object_or_404
+from .models import Task, Tag
+from .forms import TaskForm, TagForm
```
Comment (pavlopro) [inline todo/views.py:8]: Add blank line between different import groups

use absolute import instead of relative

**Review by pavlopro — CHANGES_REQUESTED** [review]

File: todo/forms.py (hunk 7 above)
Comment (pavlopro) [reply todo/forms.py]: not resolved

File: todo/models.py [hunk 10]
```diff
@@ -0,0 +1,32 @@
+from django.db import models
+
+from django.db import models
```
Comment (pavlopro) [inline todo/models.py]: double import

**Review by pavlopro — CHANGES_REQUESTED** [review]

**Review by LLkaia — COMMENTED** [review]
PR is close, is it expected?

**Pull request closed without merging**